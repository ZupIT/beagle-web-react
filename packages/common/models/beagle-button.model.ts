import { ClickEvent } from '@zup-it/beagle-web'
import { BeagleComponent } from '../types'
import { BeagleDefaultComponent } from './types'


export interface BeagleButtonInterface extends BeagleDefaultComponent, BeagleComponent {
  text: string,
  onPress?: () => void,
  clickAnalyticsEvent?: ClickEvent,
  disabled?: boolean, 
}
