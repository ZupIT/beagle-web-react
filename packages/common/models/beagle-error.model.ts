import { ErrorComponentParams } from '@zup-it/beagle-web'
import { BeagleDefaultComponent } from './types'

export interface BeagleErrorInterface extends BeagleDefaultComponent, ErrorComponentParams {
}
