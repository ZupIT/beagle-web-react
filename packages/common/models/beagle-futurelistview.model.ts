/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { BeagleUIElement } from '@zup-it/beagle-web'
import { BeagleComponent } from '../types'
import { Direction, BeagleDefaultComponent } from './types'

export type NodeType = HTMLElement | null

export interface BeagleFutureListViewInterface extends BeagleDefaultComponent, BeagleComponent {
  direction: Direction,
  dataSource: any[],
  iteratorName?: string,
  onInit?: () => void,
  onScrollEnd?: () => void,
  scrollEndThreshold?: number,
  template: BeagleUIElement,
  useParentScroll?: boolean,
  /* the property "key" is not allowed in React. Since this is not a rule for Beagle, every time
  Beagle receives "key", it transforms it into "_key" */
  _key?: string,
  __suffix__?: string,
}

export interface ScrollInterface {
  direction: Direction,
  onScrollEnd?: () => void,
  scrollEndThreshold: number,
  useParentScroll: boolean,
  elementRef: React.MutableRefObject<HTMLDivElement>,
  hasRendered: boolean,
}
