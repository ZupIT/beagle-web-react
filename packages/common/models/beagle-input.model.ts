import { BeagleDefaultComponent } from './types'

export type InputHandler = (event: { value: string }) => void

export type InputType = 'DATE' | 'EMAIL' | 'PASSWORD' | 'NUMBER' | 'TEXT'

export interface BeagleTextInputInterface extends BeagleDefaultComponent {
  value: string,
  onChange?: InputHandler,
  onFocus?: InputHandler,
  onBlur?: InputHandler,
  /**
  * @deprecated since version 1.7.0. Use `enabled` instead.
  *  Use field `enabled` to control whether the button is enabled or not in the layout.
  */
  disabled?: boolean,
  enabled?: boolean,
  readOnly?: boolean,
  placeholder?: string,
  type?: InputType,
  error?: string,
  showError?: boolean,
}

export interface BeagleTextAreaInterface extends BeagleTextInputInterface {
  name?: string,
  label?: string,
}

