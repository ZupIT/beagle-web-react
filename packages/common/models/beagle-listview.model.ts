import { BeagleDefaultComponent, Direction } from './types'

export interface BeagleListViewInterface extends BeagleDefaultComponent {
  direction: Direction,
}
