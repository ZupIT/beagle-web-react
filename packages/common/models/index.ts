/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/
export * from './beagle-image.model'
export * from './beagle-text.model'
export * from './beagle-button.model'
export * from './beagle-listview.model'
export * from './beagle-modal.model'
export * from './beagle-input.model'
export * from './beagle-container.model'
export * from './beagle-touchable.model'
export * from './beagle-tabbar.model'
export * from './beagle-pageview.model'
export * from './beagle-pageindicator.model'
export * from './beagle-lazy.model'
export * from './beagle-simpleform.model'
export * from './beagle-webview.model'
export * from './beagle-futurelistview.model'
export * from './beagle-screen.model'
export * from './beagle-error.model'
export * from './types'
