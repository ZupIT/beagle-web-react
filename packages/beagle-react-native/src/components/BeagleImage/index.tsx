/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC, useContext } from 'react'
import { ImageResizeMode, StyleSheet, Image, ImageSourcePropType } from 'react-native'
import { BeagleImageInterface } from 'common/models'
import BeagleServiceContext from 'common/provider'
import { BeagleUIReactNativeService } from 'common/types'


const modeMap: Record<string, ImageResizeMode> = {
  FIT_XY: 'stretch',
  CENTER_CROP: 'cover',
  CENTER: 'contain',
  FIT_CENTER: 'center',
}

const BeagleImage: FC<BeagleImageInterface> = props => {
  const { path, mode, accessibility, style } = props
  const beagleService = useContext(BeagleServiceContext) as BeagleUIReactNativeService
  const imgResize: ImageResizeMode = mode && modeMap[mode] || 'contain'
  
  const styleSheet = StyleSheet.create({
    fromBffStyles: {
      ...style,
    },
    defaultStyles: {
      flex:  style && style.flex ? Number(style.flex) : 1,
      resizeMode: imgResize,
      height: style && style.height || '100%',
      width: style && style.width || '100%',
    },
  })

  const localAssetsPath = beagleService?.getConfig().localAssetsPath
  let imgSource: ImageSourcePropType = {}

  if (path._beagleImagePath_ === 'local' && path.mobileId) {
    if (localAssetsPath && localAssetsPath[path.mobileId])
      imgSource = localAssetsPath[path.mobileId]
    if (!imgSource) throw new Error(`Beagle could not find image source ${path.mobileId}`)
  } else {
    imgSource = { uri: path && path.url }
  }

  return (
    <Image
      source={imgSource}
      style={{ ...styleSheet.defaultStyles, ...styleSheet.fromBffStyles }}
      {...accessibility}
    />
  )
}

export default BeagleImage
