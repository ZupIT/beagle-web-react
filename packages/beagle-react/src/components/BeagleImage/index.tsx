/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC, useContext } from 'react'
import BeagleServiceContext from 'common/provider'
import { BeagleImageInterface } from 'common/models'
import withTheme from '../utils/withTheme'
import { StyledImage, StyledFigure } from './styled'

const modeMap = {
  FIT_XY: 'fill',
  CENTER_CROP: 'none',
  CENTER: 'contain',
  FIT_CENTER: 'cover',
}

const BeagleImage: FC<BeagleImageInterface> = ({
  className,
  path,
  style,
  mode,
  accessibility,
}) => {
  const beagleService = useContext(BeagleServiceContext)
  const urlBuilder = beagleService && beagleService.urlBuilder
  const source = (path._beagleImagePath_ === 'local' || !urlBuilder)
    ? path.url
    : urlBuilder.build(path.url)

  return (
    <StyledFigure className={className} style={style} >
      <StyledImage mode={(mode && modeMap[mode]) || 'cover'}
        src={source} alt={accessibility && accessibility.accessibilityLabel} />
    </StyledFigure>
  )
}

export default withTheme(BeagleImage)
