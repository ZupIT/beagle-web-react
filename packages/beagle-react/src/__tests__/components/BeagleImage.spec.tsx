/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

// Link.react.test.js
import React from 'react'
import { mock } from 'jest-mock-extended'
import Adapter from 'enzyme-adapter-react-16'
import { configure, mount } from 'enzyme'
import { ViewContentManager } from '@zup-it/beagle-web'
import BeagleImage from '../../components/BeagleImage'
import { ImagePath } from 'common/models'

let beagleContextMock: any
let wrapper: any
const mockStyle: React.CSSProperties = {
  height: '100',
  width: '50',
}
const mockPath: ImagePath = {
  _beagleImagePath_: 'local',
  url: 'mytesteimg.png',
}

configure({ adapter: new Adapter() })

beforeAll(() => {
  beagleContextMock = mock<ViewContentManager>()
  wrapper = mount(<BeagleImage
    className="Test Class"
    style={mockStyle}
    accessibility={{accessibilityLabel: 'Test label', accessible: true  }}
    path={mockPath} />)
})

test('Beagle snapshot image', () => {
  expect(wrapper).toMatchSnapshot()
})
