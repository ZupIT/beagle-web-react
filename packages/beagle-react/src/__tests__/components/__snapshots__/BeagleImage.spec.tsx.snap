// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beagle snapshot image 1`] = `
<ComponentWithTheme
  accessibility={
    Object {
      "accessibilityLabel": "Test label",
      "accessible": true,
    }
  }
  className="Test Class"
  path={
    Object {
      "_beagleImagePath_": "local",
      "url": "mytesteimg.png",
    }
  }
  style={
    Object {
      "height": "100",
      "width": "50",
    }
  }
>
  <BeagleImage
    accessibility={
      Object {
        "accessibilityLabel": "Test label",
        "accessible": true,
      }
    }
    className="Test Class"
    path={
      Object {
        "_beagleImagePath_": "local",
        "url": "mytesteimg.png",
      }
    }
    style={
      Object {
        "height": "100",
        "width": "50",
      }
    }
  >
    <styled.figure
      className="Test Class"
      style={
        Object {
          "height": "100",
          "width": "50",
        }
      }
    >
      <figure
        className="sc-gsTCUz crUlbz Test Class"
        style={
          Object {
            "height": "100",
            "width": "50",
          }
        }
      >
        <styled.img
          alt="Test label"
          mode="cover"
          src="mytesteimg.png"
        >
          <img
            alt="Test label"
            className="sc-bdfBwQ ednZQG"
            mode="cover"
            src="mytesteimg.png"
          />
        </styled.img>
      </figure>
    </styled.figure>
  </BeagleImage>
</ComponentWithTheme>
`;
